### DISCUSSION TITLE
How to set headers to public assets?

### DISCUSSION THREAD

**Discussion by peterkimzz** [nuxt/nuxt#23012] [Questions]
Hi,

I'm trying to load a javascript file by pressing a "load button", but it doesn't have headers that I set in server middleware.

```ts
// ~server/middleware/header.ts

import { H3Event } from 'h3';

// This header setting is for using wasp in a browser.
export default defineEventHandler((event: H3Event) => {
  event.node.res.setHeader('Access-Control-Allow-Origin', '*');
  event.node.res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp');
  event.node.res.setHeader('Cross-Origin-Opener-Policy', 'same-origin');
})
```

<img width="791" alt="image" src="https://github.com/nuxt/nuxt/assets/20244536/f8fc8cee-15e7-4810-97ab-324798b33cb9">

### Here is the thing that I want

<img width="790" alt="image" src="https://github.com/nuxt/nuxt/assets/20244536/8da52f51-8368-48ae-81bd-409080fe9ad4">

**Comment by xjccc** [comment]
https://nuxt.com/docs/guide/directory-structure/server#return-a-legacy-handler-or-middleware

use this to try?

**Reply by peterkimzz** [reply]
@xjccc I tried the legacy middleware, but doesn't work.

I've resolved this issue with setting headers to `vercel.json`. Yes, it is not a solution from nuxt framework, so we still need better solution by using nuxt.

https://vercel.com/docs/projects/project-configuration#headers

**Reply by xjccc** [reply]
emmm.. i use `fromNodeMiddleware`, res.setHeader('some thing', 'xxx') is useful, I'm not sure if this will work for you

**Reply by peterkimzz** [reply]
Right it's useful and here is my code, but got same results.

```ts
export default fromNodeMiddleware((req, res, next) => {
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader("Cross-Origin-Embedder-Policy", "require-corp");
  res.setHeader("Cross-Origin-Opener-Policy", "same-origin");

  next();
});
```

**Reply by xjccc** [reply]
try this?

```
nitro: {
   '/assets/*': {
      headers: {
         "Access-Control-Allow-Origin": "*"
      }
   }
}

```

**Comment by Birleyici** [comment]
Is there any progress on this issue? Did you find a solution?
